### DISCUSSION TITLE
API for retrieving additional parameters from httpx.Cookies

### DISCUSSION THREAD

**Discussion by jackm** [encode/httpx#1497] [Q&A] [answered]
With regard to a `httpx.Cookies` object, how can I get other cookie parameters aside from its value?
`httpx.Cookies.get()` retrieves the cookie value, but how could I get say the expiry date as well? Is there a user accessible API for this? I did not see anything about this in the httpx documentation.

Relevant issue #1509

**Comment by lundberg** [comment, marked as answer]
There's no current way of retrieving this that I can see either, other than manually looking through the cookies in the `jar`, but we shouldn't do that IMO.

But do we really need an api to get these kind of parameters?

Maybe it's nicer if parameters like `Expires` and `Max-Age` was handled/respected in `httpx.Cookies.get()`, i.e. not returning a value for an expired cookie or an `HttpOnly` one.

Not sure myself, just some thoughts 😉

**Comment by nidico** [comment]
In my opinion having an API for cookie expire time access is required.

I'm migrating from requests to httpx following [starlette 0.21.0](https://github.com/encode/starlette/releases/tag/0.21.0). In my case, I want to check in a test case whether `expires` is set correctly.

Accessing `response.cookies.jar._cookies["testserver.local"]["/"]` is possible but seems to be an internal, undocumented implementation detail and parsing `response.headers["set-cookie"]` isn't particularly pleasant either...

**Reply by lovelydinosaur** [reply]
What would you suggest?

> I'm migrating from requests to httpx

Are there differences here that make it either easier or more difficult with `httpx`?

**Reply by nidico** [reply]
> Are there differences here that make it either easier or more difficult with `httpx`?

After looking at this with a clear head again, it doesn't make much of a difference.

In both libraries I need to go through a `http.cookiejar.CookieJar` compatible object out of which I need to get the right `http.cookiejar.Cookie` in one way or the other, e.g. through `next(c for c in response.cookies if c.name == NAME).expires`. (Calling `response.cookies.jar._cookies["testserver.local"]["/"]` - as I first did - is not needed!)

`requests` is *a bit* easier here, as `requests.models.Response.cookies` directly exposes a `requests.cookies.RequestsCookieJar` which is compatible to the default `http.cookiejar.CookieJar`, but also allows dict access like `httpx.Cookies` at the same time, whereas `httpx.Cookies` provides the dict access and `httpx.Cookies.jar` exposes the `CookieJar` API.

> What would you suggest?

Nothing, really. It's a bit unfortunate that the httpx API differs in this case and would prefer if it was just the same, but it probably is different for a good reason and it's not much of a hassle to just call `.jar` on `response.cookies`.
